app/views/users/show.html.erb

```diff
@@ -6,7 +6,7 @@
 
 <!-- hide private users profile -->
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show? %>
```

Comment:
Noticed how we can't even send a follow request to a private user? You may want to dissociate the policy here from the `show?` policy. We want to be able to show a user's profile even if they are private, but hide their photo.